scripts/run_e2e_tests.py

```diff
@@ -585,6 +583,8 @@ def main(args=None):
             break
         # Prepare for rerun.
         cleanup()
+        for port in [OPPIA_SERVER_PORT, GOOGLE_APP_ENGINE_PORT]:
```

Comment:
Should we do this in the cleanup function? (We are also doing `stop_redis_server` inside that function which seems much reasonable to me.)

Reply:
I don't think so. The `cleanup` function always runs at the ends of tests, but we don't need to wait for shutdown after the last run. We only need to wait when we are about to do another rerun. For example, we don't need to wait if the test passes, right?

Reply:
> We only need to wait when we are about to do another rerun. For example, we don't need to wait if the test passes, right?

I think we should stop the script after cleaning up everything irrespective of whether the test passes or not. (Currently, we do wait for the Redis server to stop even if the test passes.)

Also, this script is also being used by contributors to run tests locally we should do the cleanup. wdyt?

Reply:
Done in 2ce96489d